### PR TITLE
Issue 332 generic parallel containers

### PR DESCRIPTION
**Pull request by VladimirStepanov**
Close https://github.com/testcontainers/testcontainers-go/issues/332

### CODE REVIEW

File: generic.go
Comment (mdelapenya) [inline generic.go:9]: Just a personal opinion, not a blocker: wdyt about moving all this new code to a `parallel.go`, or `generic_parallel.go` file, so that all the concerns about parallelisation are all together in one same file? I'd like to listen to your thoughts, and if you see it convenient to keep the code next to the generic implementation, what would it be the motivation?

File: parallel.go
Comment (VladimirStepanov) [reply parallel.go]: Yeah) good point

**Comment by mdelapenya** [comment]
I left a few comments, more cosmetic than technical, so I'd like to thank you for dedicating your time to contribute this PR. Technically speaking, it seems perfect for the use case.

**Comment by VladimirStepanov** [comment]
@mdelapenya I added all suggestions to code, plz check if you have free minute:)

File: docs/features/creating_container.md
Comment (mdelapenya) [inline docs/features/creating_container.md:88]:
```suggestion

The following test creates two NGINX containers in parallel:

```

File: parallel.go
Comment (mdelapenya) [inline parallel.go:40]: Readability :)
```suggestion

	for req := range requests {
```

File: parallel_test.go
Comment (mdelapenya) [inline parallel_test.go:39]: Although the default int value is zero, I'd add it here explicitly for readability:
```suggestion
			resLen: 2,
			expErrors: 0,
```

File: parallel_test.go
Comment (mdelapenya) [inline parallel_test.go:42]: Can we add another test entry for two errors to cover maxLength of the container requests?

**Review by mdelapenya — APPROVED** [review]
@VladimirStepanov this LGTM, great contribution!

I left a few comments, not blocking the PR that's why I'm approving it before hand, but would like you to review them. before merging it.

Thanks again for your time!

**Comment by mdelapenya** [comment]
Thank you @VladimirStepanov! This PR is wonderful!

**Pull request merged**
